sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/TopicSubscriptionWithRuleOperationsSample.java

```diff
@@ -0,0 +1,222 @@
+        // Delete Default Rule.
+        // Add the required SqlFilter Rule
+        // Note: Does not apply to this sample but if there are multiple rules configured for a
+        // single subscription, then one message is delivered to the subscription when any of the
+        // rule matches. If more than one rules match and if there is no `SqlRuleAction` set for the
+        // rule, then only one message will be delivered to the subscription. If more than one rules
+        // match and there is a `SqlRuleAction` specified for the rule, then one message per `SqlRuleAction`
+        // is delivered to the subscription.
+        System.out.println(String.format("SubscriptionName: %s, Removing Default Rule and Adding SqlFilter", SQL_FILTER_ONLY_SUBSCRIPTION_NAME));
+        administrationAsyncClient.deleteRuleWithResponse(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_ONLY_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME).block();
+        administrationAsyncClient.createRuleWithResponse(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_ONLY_SUBSCRIPTION_NAME, SQL_FILTER_ONLY_SUBSCRIPTION_RULE_NAME,
+            new CreateRuleOptions(new SqlRuleFilter("Color = 'Red'"))).block();
+
+        // 3rd Subscription: Add SqlFilter and SqlRuleAction on Subscription 3
+        // Delete Default Rule
+        // Add the required SqlFilter Rule and Action
+        System.out.println(String.format("SubscriptionName: %s, Removing Default Rule and Adding CorrelationFilter", SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME));
+        administrationAsyncClient.deleteRuleWithResponse(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME).block();
+        administrationAsyncClient.createRuleWithResponse(SERVICE_BUS_TOPIC_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME, SQL_FILTER_WITH_ACTION_SUBSCRIPTION_RULE_NAME,
+            new CreateRuleOptions().setFilter(new SqlRuleFilter("Color = 'Blue'"))
+                .setAction(new SqlRuleAction("SET Color = 'BlueProcessed'"))).block();
+
+        // 4th Subscription: Add Correlation Filter on Subscription 4
+        System.out.println(String.format("SubscriptionName: %s, Removing Default Rule and Adding CorrelationFilter", CORRELATION_FILTER_SUBSCRIPTION_NAME));
+        administrationAsyncClient.deleteRuleWithResponse(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME).block();
+        administrationAsyncClient.createRuleWithResponse(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME, CORRELATION_FILTER_SUBSCRIPTION_RULE_NAME,
+            new CreateRuleOptions(new CorrelationRuleFilter().setCorrelationId("important").setLabel("Red"))).block();
+
+        // Get Rules on Subscription, called here only for one subscription as example
+        administrationAsyncClient.listRules(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME).collectList()
+            .doOnSuccess(listRuleProperties -> {
```

Comment:
We usually don't use side-effect APIs like `doOn...` to process data. If you change to use sync client, this isn't relevant any more.

Reply:
According to your comment, fixed in new version.